controller/controller_server/utils.py

```diff
@@ -151,18 +157,47 @@ def validate_create_snapshot_request(request):
 def validate_delete_snapshot_request(request):
     logger.debug("validating delete snapshot request")
     if not request.snapshot_id:
-        raise ValidationException(messages.name_should_not_be_empty_message)
-    logger.debug("validating secrets")
-    if request.secrets:
-        validate_secret(request.secrets)
+        raise ValidationException(messages.snapshot_id_should_not_be_empty_message)
+
+    validate_secrets(request.secrets)
+
     logger.debug("request validation finished.")
 
 
+def validate_validate_volume_capabilities_request(request):
+    logger.debug("validating validate_volume_capabilities request")
+
+    logger.debug("validating volume id")
+    volume_id = request.volume_id
+    if not volume_id:
+        raise ValidationException(messages.volume_id_should_not_be_empty_message)
+    if config.PARAMETERS_OBJECT_ID_DELIMITER not in volume_id:
+        raise ObjectIdError(config.VOLUME_TYPE_NAME, volume_id)
+
+    logger.debug("validating parameters")
```

Comment:
Put inside the `_validate_pool_parameter` function

Reply:
if so then change string specifically to pool parameter (as the function name)

Reply:
done